packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts

```diff
@@ -225,6 +339,42 @@ export class AppSyncModelJavaVisitor<
         ['Override']
       );
     });
+
+    // Add id builder
+    const idBuildStepBody = dedent`this.id = id;
+
+    try {
+        UUID.fromString(id); // Check that ID is in the UUID format - if not an exception is thrown
+    } catch (Exception exception) {
+        throw new AmplifyException("Model IDs must be unique in the format of UUID.",
+                exception,
+                "If you are creating a new object, leave ID blank and one will be autogenerated for you. " +
+                "Otherwise, if you are referencing an existing object, be sure you are getting the correct " +
+                "id for it. It's also possible you are referring to an item created outside of Amplify. We " +
+                "currently do not support this.",
+                false);
+    }
+
+    return this;`;
+
+    const idComment = dedent`WARNING: Do not set ID when creating a new object. Leave this blank and one will be auto generated for you.
```

Comment:
"auto generated" vs. "autogenerated". Above "autogenerated" is used, please make sure we're consistent.